[jsk_pcl_ros] set tracking model with marker in particle_filter_tracker

aginika opened the pull request:
DONT MERGE YET

k-okada commented:
please add example or documents too

--
◉ Kei Okada


On Thu, Jul 16, 2015 at 11:10 PM, Yuto Inagaki <notifications@github.com>
wrote:

> DONT MERGE YET
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/jsk-ros-pkg/jsk_recognition/pull/1006
> Commit Summary
>
>    - [jsk_pcl_ros] set tracking model with marker in
>    particle_filter_tracker
>
> File Changes
>
>    - *M* jsk_pcl_ros/include/jsk_pcl_ros/particle_filter_tracking.h
>    <https://github.com/jsk-ros-pkg/jsk_recognition/pull/1006/files#diff-0>
>    (5)
>    - *M* jsk_pcl_ros/include/jsk_pcl_ros/pcl_conversion_util.h
>    <https://github.com/jsk-ros-pkg/jsk_recognition/pull/1006/files#diff-1>
>    (59)
>    - *M* jsk_pcl_ros/src/particle_filter_tracking_nodelet.cpp
>    <https://github.com/jsk-ros-pkg/jsk_recognition/pull/1006/files#diff-2>
>    (20)
>
> Patch Links:
>
>    - https://github.com/jsk-ros-pkg/jsk_recognition/pull/1006.patch
>    - https://github.com/jsk-ros-pkg/jsk_recognition/pull/1006.diff
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jsk-ros-pkg/jsk_recognition/pull/1006>.
>

aginika commented:
@k-okada 
Thank you. I added sample and documents.

I also move the target according to the marker.position
and
I added color too.

Now, user can execute demonstration with below code if openni topic and tf is correctly published.
```
roslaunch jsk_pcl_ros tracking.launch _track_with_model_sample:=true
```

aginika commented:
Please merge if this passed. 
Thanks

garaemon commented on jsk_pcl_ros/src/particle_filter_tracking_nodelet.cpp:
```diff
@@ -394,6 +398,31 @@ namespace jsk_pcl_ros
+                            );
+
+      Eigen::Affine3d trans_3d; Eigen::Affine3f trans;
+      tf::poseMsgToEigen(marker.pose, trans_3d);
```
if you include pcl_conversion_util.h, you can use tf::poseMsgToEigen with Eigen::Affine3f

garaemon commented on jsk_pcl_ros/src/particle_filter_tracking_nodelet.cpp:
```diff
@@ -394,6 +398,31 @@ namespace jsk_pcl_ros
+      frame_id_ = marker.header.frame_id;
+      reset_tracking_target_model(cloud);
+    }else{
+      ROS_ERROR(" Marker Models type is not TRIANGLE ");
```
Use `JSK_ROS_ERROR`

aginika commented:
@garaemon Thank you for your advise.
I modified code.
